pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -367,11 +386,18 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod) {
 		// update tr completed time
 		taskRun.Status.CompletionTime = &metav1.Time{Time: time.Now()}
 	case corev1.PodPending:
-		msg := getWaitingMessage(pod)
+		var reason, msg string
+		if isPodExceedingNodeResources(pod) {
+			reason = reasonExceededNodeResources
+			msg = getExceededResourcesMessage(taskRun)
+		} else {
+			reason = "Pending"
+			msg = getWaitingMessage(pod)
+		}
```

Comment:
it feels like we're starting to build up a lot of logic in this file/package around looking at the pod and from that determining a `reason` and a `msg` - I wonder if we could move some of this out into its own package, with tests?

anyway i dont feel strongly enough about this to block merging tho, maybe it's something we can revisit